flake-modules/cuda.nix

```diff
@@ -30,31 +32,36 @@ topLevel@{ flake-parts-lib, inputs, ... }: {
             type = lib.types.package;
             default = pkgs.symlinkJoin {
               name = "cuda-home";
-              paths = common.config.cuda.packages;
+              paths = finalPackages;
             };
           };
-          options.cuda.packages = lib.mkOption {
-            type = lib.types.listOf lib.types.package;
+          options.cuda.version = lib.mkOption {
```

Comment:
```suggestion
          options.cuda.cudaPackages = lib.mkOption {
```

Reply:
I think it would be difficult for the user to arbitrarily select the packages included in cudaPackages with the proposed method, what do you think?
Using functions in modules is common in NixOSModule, although I think it's a bit difficult if someone are not familiar with Nix.

Reply:
> I think it would be difficult for the user to arbitrarily select the packages included in cudaPackages with the proposed method

I don't understand why would be difficult. Could you explain more?

Reply:
"Difficult to understand" was correct.
Although it is trivial, I don't think it can be declared without using rec.
```nix
common.cuda = rec {
  cudaPackages = pkgs.cudaPackages_12_1;
  packages = with cudaPackages; [
    ...
  ];
};
```

Reply:
rec is unnecessary. See the suggested change.

Reply:
Doesn't it take into account the unique definition of the package? I'm sorry if I'm wrong.
I think it would be a good idea to delete cuda.cudaPackages and define only cuda.packages.

- miss
  ![image](https://github.com/user-attachments/assets/13b86bcc-38cd-4233-bf28-953cc36861a0)

- correct
  ![image](https://github.com/user-attachments/assets/e70fdfef-a50a-49e7-ae8f-e64ce6b18884)

Reply:
Try `with common.config.cuda.cudaPackages;`

Reply:
A more common style is to avoid `with`. Instead, do something like
```
let cfg = common.config.cuda; in {
  cudaPackages = pkgs.cudaPackages_12_1;
  packages = [ cfg.cudaPackages.cudnn cfg.cudaPackages.cuda_nvcc ];
}

```

Reply:
You can find examples at https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+cfg+%3D&type=code